### PR TITLE
[type-declarations] Add return type declaration based on constants + add known array types

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -34,7 +34,6 @@ parameters:
...
 		objectManagerLoader: tests/object-manager.php
 		allowNullablePropertyForRequiredField: true
-	symfony:
-		containerXmlPath: var/cache/test/AppKernelTestDebugContainer.xml
-		consoleApplicationLoader: app/console-application.php
-		
\ No newline at end of file
+#	symfony:
+#		containerXmlPath: var/cache/test/AppKernelTestDebugContainer.xml
+#		consoleApplicationLoader: app/console-application.php
```
Comment (escopecz) [inline phpstan.neon]: Was this change intentional? If so, what was the reason to disable it?

File: rector.php [hunk 2]
```diff
@@ -16,7 +17,19 @@
...
+        ReturnTypeFromReturnDirectArrayRector::class => [
+            // require bit test update
+            __DIR__.'/app/bundles/LeadBundle/Model/LeadModel.php',
+            // skip as a placeholder, not bool really
+            __DIR__.'/plugins/MauticCrmBundle/Integration/ZohoIntegration.php',
+        ],
+
+        \Rector\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictBoolReturnExprRector::class => [
+            // skip as a placeholder, not bool really
+            __DIR__.'/app/bundles/LeadBundle/Segment/Decorator/BaseDecorator.php',
```
Comment (escopecz) [inline rector.php]: Are these comments up to date?

Comment (TomasVotruba) [reply rector.php]: This PR is still WIP. Gotta run now.

I'll ping you once it's ready for review :+1:

File: phpstan.neon [hunk 3]
```diff
@@ -28,6 +28,8 @@ parameters:
 		- phpstan-bootstrap.php
 	ignoreErrors:
 		- '/Variable \$\w+ might not be defined\./'
+		# resolve later with translator interface generics
+		- '#Method Mautic\\PageBundle\\Model\\PageModel::getTranslatedEntity\(\) return type has no value type specified in iterable type array#'
```
Comment (TomasVotruba) [inline phpstan.neon]: I put the ignored error here on purpose, so we are aware of the need for generics with translator. Hope it's ok

**Comment by TomasVotruba** [comment]
Finally :tada:  :sweat_smile: , good to go :+1: 

I'll collect a lot of feedback for Rector, to improve the rules to work better :pray:

File: .php-cs-fixer.php [hunk 4]
```diff
@@ -9,7 +9,8 @@
     ->in(__DIR__.'/app/middlewares')
     ->in(__DIR__.'/app/migrations')
     ->in(__DIR__.'/plugins')
-    ->in(__DIR__.'/.github/workflows/mautic-asset-upload');
+    ->in(__DIR__.'/.github/workflows/mautic-asset-upload')
+    ->append([__DIR__.'/rector.php']);
```
Comment (TomasVotruba) [inline .php-cs-fixer.php:13]: This will save us lot of time from CI debug while changing dev configs :+1:

File: app/bundles/LeadBundle/Model/LeadModel.php [hunk 5]
```diff
@@ -2174,7 +2174,7 @@ public function getPreferredChannel(Lead $lead)
     }
 
     /**
-     * @return array
+     * @return mixed[]
      */
     public function setPrimaryCompany($companyId, $leadId)
```
Comment (escopecz) [inline app/bundles/LeadBundle/Model/LeadModel.php:2179]: Rector missed this one 😄

File: app/bundles/PluginBundle/Integration/AbstractIntegration.php [hunk 6]
```diff
@@ -242,10 +242,8 @@ abstract public function getAuthenticationType();
 
     /**
      * Get if data priority is enabled in the integration or not default is false.
-     *
-     * @return string
      */
-    public function getDataPriority()
+    public function getDataPriority(): bool
```
Comment (escopecz) [inline app/bundles/PluginBundle/Integration/AbstractIntegration.php:246]: I double-checked that all integrations shipped with the core and extending this class do have the bool return type in this or previous PRs added. Otherwise this would throw a 500 error. 👍

**Review by escopecz — APPROVED** [review]
I didn't spot any issue. Nice improvement. Thanks Tomas! 👍

File: app/bundles/LeadBundle/Model/LeadModel.php (hunk 5 above)
Comment (TomasVotruba) [reply app/bundles/LeadBundle/Model/LeadModel.php:2179]: This was actually fixed by me manually, just to silence PHPStan 😆

Setter with return value should be fixed to do one thing only.
